filemapping_unix.go

```diff
@@ -1,15 +1,22 @@
+//go:build !windows
 // +build !windows
 
 package gettext
 
 import (
+	"errors"
+	"io/fs"
 	"os"
 	"syscall"
 )
 
-func (m *fileMapping) tryMap(f *os.File, size int64) error {
+func (m *fileMapping) tryMap(f fs.File, size int64) error {
 	var err error
-	m.data, err = syscall.Mmap(int(f.Fd()), 0, int(size), syscall.PROT_READ, syscall.MAP_PRIVATE)
+	var of, ok = f.(*os.File)
```

Comment:
(super-nitck) `of, ok := f.(*os.File)` (without the var) seems to be the slightly more idiomatic way to do this :)